Source/WTF/wtf/URL.cpp

```diff
@@ -843,6 +843,28 @@ String URL::strippedForUseAsReferrer() const
     );
 }
 
+String URL::strippedForUseAsReferrerWithExplicitPort() const
```

Comment:
Why are we adding this new function? Please explain in the commit message.

Reply:
Will do. In short, WPT expects the port name be included in the URL supplied to the report. Perhaps a better name would be "URL::stringSuitableForReporting" or "URL::stringIncludingExplicitPort".

Reply:
Huh, that's so bizarre. Is that mentioned anywhere in the spec?